### PR TITLE
Prevent compression of binary formats

### PR DESCRIPTION
**Pull request by ArielYehezkely**
Prevent compression when formats are already binary.
Fix device authentication

### CODE REVIEW

File: azure-kusto-data/azure/kusto/data/security.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def __init__(self, kcsb):
         else:
             self._authentication_method = AuthenticationMethod.aad_device_login
             self._client_id = "db662dc1-0cfe-4e1c-a843-19a68e65be58"
-            create_adal_context = False
+            create_adal_context = True
```
Comment (ohadbitt) [inline azure-kusto-data/azure/kusto/data/security.py]: are you sure ?

File: azure-kusto-ingest/azure/kusto/ingest/_descriptors.py [hunk 2]
```diff
@@ -38,24 +32,12 @@ def __init__(self, path, size=0, source_id=None):
         assert_uuid4(source_id, "source_id must be a valid uuid4")
         self.source_id = source_id
         self.stream_name = os.path.basename(self.path)
-        if self.path.endswith(".gz") or self.path.endswith(".zip"):
-            self.zipped_stream = open(self.path, "rb")
-            if not self.size or self.size <= 0:
+        if not self.size or self.size <= 0:
+            if self.path.endswith(".gz") or self.path.endswith(".zip"):
                 # TODO: this can be improved by reading last 4 bytes
                 self.size = int(os.path.getsize(self.path)) * 5
```
Comment (ohadbitt) [inline azure-kusto-ingest/azure/kusto/ingest/_descriptors.py]: why isnt it 11 as in all our sdks?

File: azure-kusto-ingest/azure/kusto/ingest/_descriptors.py [hunk 3]
```diff
@@ -38,24 +32,12 @@ def __init__(self, path, size=0, source_id=None):
...
-            self.zipped_stream = BytesIO()
-            with open(self.path, "rb") as f_in, GzipFile(filename="data", fileobj=self.zipped_stream, mode="wb") as f_out:
-                shutil.copyfileobj(f_in, f_out)
-            self.zipped_stream.seek(0)
-
-    def delete_files(self):
-        """Deletes the gz file if the original file was not zipped.
-        In case of success deletes the original file as well."""
-        if self.zipped_stream is not None:
-            self.zipped_stream.close()
```
Comment (ohadbitt) [inline azure-kusto-ingest/azure/kusto/ingest/_descriptors.py]: why?

File: azure-kusto-ingest/tests/test_descriptors.py [hunk 4]
```diff
@@ -1,60 +0,0 @@
-import sys
-from os import path
-import unittest
-from azure.kusto.ingest import FileDescriptor
-
-
-class DescriptorsTest(unittest.TestCase):
```
Comment (ohadbitt) [inline azure-kusto-ingest/tests/test_descriptors.py]: why?

File: azure-kusto-data/azure/kusto/data/security.py (hunk 1 above)
Comment (ArielYehezkely) [reply azure-kusto-data/azure/kusto/data/security.py]: Yes, the previous change disabled the device authentication by mistake.

File: azure-kusto-ingest/azure/kusto/ingest/_descriptors.py (hunk 3 above)
Comment (ArielYehezkely) [reply azure-kusto-ingest/azure/kusto/ingest/_descriptors.py]: I don't keep the stream inside the descriptor anymore - therefore it's not its responsibility to close it.

File: azure-kusto-ingest/azure/kusto/ingest/_descriptors.py (hunk 2 above)
Comment (ArielYehezkely) [reply azure-kusto-ingest/azure/kusto/ingest/_descriptors.py]: Frankly I don't know..

File: azure-kusto-ingest/azure/kusto/ingest/_ingest_client.py [hunk 5]
```diff
@@ -46,40 +50,39 @@ def ingest_from_dataframe(self, df, ingestion_properties):
...
         containers = self._resource_manager.get_containers()
 
         if isinstance(file_descriptor, FileDescriptor):
             descriptor = file_descriptor
         else:
             descriptor = FileDescriptor(file_descriptor)
 
-        file_descriptors.append(descriptor)
+        if (
+            ingestion_properties.format == DataFormat.PARQUET
```
Comment (danield137) [inline azure-kusto-ingest/azure/kusto/ingest/_ingest_client.py]: `if ingestion_properties.format in [DataFormat.PARQUET, ...]`

File: azure-kusto-ingest/azure/kusto/ingest/_ingest_client.py [hunk 6]
```diff
@@ -46,40 +50,39 @@ def ingest_from_dataframe(self, df, ingestion_properties):
...
+            or ingestion_properties.format == DataFormat.ORC
+            or ingestion_properties.format == DataFormat.AVRO
+            or descriptor.path.endswith(".gz")
+            or descriptor.path.endswith(".zip")
+        ):
+            zipped_stream = open(descriptor.path, "rb")
```
Comment (danield137) [inline azure-kusto-ingest/azure/kusto/ingest/_ingest_client.py]: it's not zipped stream anymore.. just file_stream

File: azure-kusto-ingest/azure/kusto/ingest/_ingest_client.py [hunk 7]
```diff
@@ -46,40 +50,39 @@ def ingest_from_dataframe(self, df, ingestion_properties):
...
+        else:
+            descriptor.stream_name += ".gz"
```
Comment (danield137) [inline azure-kusto-ingest/azure/kusto/ingest/_ingest_client.py]: I would extract this entire method..it has a very specific job, to read data from the descriptor (might as well be on the descriptor)

File: azure-kusto-ingest/azure/kusto/ingest/_ingest_client.py [hunk 8]
```diff
@@ -88,11 +91,14 @@ def ingest_from_file(self, file_descriptor, ingestion_properties):
...
         blob_service = storage_client.create_block_blob_service()
 
-        blob_service.create_blob_from_stream(container_name=container_details.object_name, blob_name=blob_name, stream=descriptor.zipped_stream)
+        blob_service.create_blob_from_stream(container_name=container_details.object_name, blob_name=blob_name, stream=zipped_stream)
         url = blob_service.make_blob_url(container_details.object_name, blob_name, sas_token=container_details.sas)
 
         self.ingest_from_blob(BlobDescriptor(url, descriptor.size, descriptor.source_id), ingestion_properties=ingestion_properties)
 
+        if zipped_stream is not None:
+            zipped_stream.close()
```
Comment (danield137) [inline azure-kusto-ingest/azure/kusto/ingest/_ingest_client.py]: this is cleaner to combine with a `with` statement

**Pull request merged**